components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -323,12 +403,21 @@ public boolean doPostGetUserClaimValue(String userName, String claim, List<Strin
         if (isEnable()) {
             JSONObject dataObject = new JSONObject();
             dataObject.put(ListenerUtils.CLAIM_URI_FIELD, claim);
-            if (CollectionUtils.isNotEmpty(claimValue)) {
+            String target = ListenerUtils.getEntityWithUserStoreDomain(userName, storeManager);
+            Map<String, String> claims = new HashMap<>();
+            if (LoggerUtils.isLogMaskingEnable && !claimValue.isEmpty()) {
+                for (String claimVal : claimValue) {
+                    claims.put(claim, claimVal);
```

Comment:
not clear on the approach taken here. can you explain?

Reply:
Here the List of claim values is mapped to the same claim URI. To match with the maskClaimValues methods' input format created a map with claim URI and claim values.